src/lightning_lite/utilities/types.py

```diff
@@ -60,18 +60,9 @@ def rank(self) -> int:
         ...
 
 
-# Inferred from `torch.optim.lr_scheduler.pyi`
-# Missing attributes were added to improve typing
-@runtime_checkable
-class _LRScheduler(_Stateful[str], Protocol):
-    optimizer: Optimizer
-    base_lrs: List[float]
-
-    def __init__(self, optimizer: Optimizer, *args: Any, **kwargs: Any) -> None:
-        ...
-
-    def step(self, epoch: Optional[int] = None) -> None:
-        ...
+LRScheduler = (
+    torch.optim.lr_scheduler.LRScheduler if _TORCH_GREATER_EQUAL_1_14 else torch.optim.lr_scheduler._LRScheduler
```

Comment:
shouldn't it be "else _LRScheduler" the one we had defined above?

Reply:
oops, will fix it asap.

Reply:
I think we need to have a `_TORCH_LRSCHEDULER` variable that points to either of these 2

And our `LRScheduler` protocol that we still use for instance checks.

Reply:
That means you want to assign `_TORCH_LRSCHEDULER` like this and import it everywhere else. RIght?
https://github.com/qmaruf/lightning/blob/feature-15587/src/lightning_lite/utilities/types.py#L75

Reply:
Yes. Note that in some places you want to use that variable, and in others the

```python
@runtime_checkable
class _LRScheduler(_Stateful[str], Protocol):
    ...
```

That was already defined. The latter mainly for `isinstance` checks. You can learn more about protocols in https://peps.python.org/pep-0544/